api/docs/scatter_gather_emulation.dox

```diff
@@ -71,6 +74,126 @@ register `k1`. Elements may be scattered in any order. When an element is stored
+handles them in a similar way.
+
+All scatter/gather/predicated contiguous instructions access elements conditionally based
+on the mask value of a governing predicate register. Loads are always zeroing: inactive
+elements are set to 0 in the destination register.
+
+### Scalar+vector
+
+```
+ld1b (%x0,%z0.s,uxtw)[1byte] %p0/z -> %z1.s
+```
+
+Above is a scalar+vector gather load that reads 8-bit values which are zero-extended and
+written to 32-bit elements of the `z1` vector register. Addresses are calculated by
+adding the base address from the scalar `x0` register to the corresponding 32-bit element
+in the vector index register `z0`. Elements that are not active in the mask contained by
+the governing predicate register `p0` are not loaded and the corresponding element in the
+destination register `z1` is set to `0`.
+
+```
+st1h   %z17.d %p5 -> (%x19,%z20.d)[2byte]
+```
+
+Above is a scalar+vector scatter store that writes the lowest 16-bits of the 64-bit
+elements of the `z17` vector register. Addresses are calculated by adding the base
+address from the scalar `x19` register to the corresponding 64-bit element in the vector
+index register `z20`. Elements that are not active in the mask contained by the governing
+predicate register `p5` are not stored.
+
+### Vector+immediate
+
+```
+ld1sw  +0x18(%z8.d)[4byte] %p2/z -> %z6.d
+```
+Above is a vector+immediate gather load that reads 32-bit values which are sign-extended
+and written to 64-bit elements of the `z6` vector register. Addresses are calculated by
+adding the immedate value 0x18 to the corresponding 64-bit element in the vector base
+register `z8`. Elements that are not active in the mask contained by governing predicate
+register `p2` are zeroed.
+
+### Vector+scalar
+
+```
+stnt1d %z27.d %p7 -> (%z25.d,%x29)[8byte]
+```
+
+Introduced with SVE2. The above instruction is a scatter store that writes the 64-bit
+elements of the `z27` vector register. Addresses are calculated by adding the value of
+scalar register `x29` to the corresponding 64-bit element in the vector base register
+`z25`. Elements that are not active in the mask contained by the governing predicate
+register `p7` are not stored.
+
+### Scalar+scalar
+
+```
+ld1sb  (%x17,%x18)[1byte] %p5/z -> %z16.d
+```
+
+The above instruction is a scalar+scalar predicated contiguous load. 8-bit values are
+loaded and sign extended to 64-bits and written to the vector register `z16`. The address
+of the first element is calculated by adding the value of the scalar index register `x18`
+to the value of the scalar base register `x17`. Addresses for subsequent elements are
+calculated by adding the size of the loaded value (1-byte in this example) to the address
+of the previous element. Elements that are not active in the mask contained by governing
+predicate register `p5` are zeroed.
+
+### Scalar+immediate
+
+```
+st1w   %z10.s %p3 -> -0x60(%x11)[4byte]
```

Comment:
Could you remind me what ".s" stands for? single-precision floats?

Reply:
`.b` byte
`.h` halfword (16-bit)
`.s` singleword (32-bit)
`.d` doubleword (64-bit)
`.q` quadword (128-bit) (though none of the scatter/gather instructions here use quadword)